tests/tests_pytorch/core/test_results.py

```diff
@@ -11,36 +11,40 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from functools import partial
+
 import torch
 import torch.distributed as dist
-import torch.multiprocessing as mp
 
-import tests_pytorch.helpers.utils as tutils
-from lightning_lite.utilities.distributed import sync_ddp_if_available
+from lightning_lite.plugins.environments import LightningEnvironment
+from pytorch_lightning.accelerators import CPUAccelerator, CUDAAccelerator, MPSAccelerator
+from pytorch_lightning.strategies import DDPSpawnStrategy
+from pytorch_lightning.strategies.launchers import _MultiProcessingLauncher
 from pytorch_lightning.trainer.connectors.logger_connector.result import _Sync
 from tests_pytorch.helpers.runif import RunIf
+from tests_pytorch.models.test_tpu import wrap_launch_function
 
 
-def _setup_ddp(rank, worldsize):
-    import os
-
-    os.environ["MASTER_ADDR"] = "localhost"
-
-    # initialize the process group
-    dist.init_process_group("gloo", rank=rank, world_size=worldsize)
+def spawn_launch(fn, parallel_devices):
```

Comment:
is this meant to be defined in core/test_results?

Reply:
No, it could be moved. I didn't put much thought into it since we don't have circular dependency issues in tests and they are already a bit chaos. If you have a specific suggestion here I'm happy to mo change it.

Reply:
I thought maybe it could originate from the launcher test file

Reply:
Well, the launcher test file does not use this. It'd be weird to have it there. We can always reorganize the tests later though.